diff -up yum-3.2.29/yum/__init__.py.old yum-3.2.29/yum/__init__.py
--- yum-3.2.29/yum/__init__.py.old	2014-05-15 11:03:27.000000000 +0200
+++ yum-3.2.29/yum/__init__.py	2014-05-15 11:09:07.838765708 +0200
@@ -805,7 +805,9 @@ class YumBase(depsolve.Depsolve):
         self.verbose_logger.log(logginglevels.DEBUG_4,
                                 _('Getting group metadata'))
         reposWithGroups = []
-        self.repos.doSetup()
+        #  Need to make sure the groups data is ready to read. Really we'd want
+        # to add groups to the mdpolicy list of the repo. but we don't atm.
+        self.pkgSack
         for repo in self.repos.listGroupsEnabled():
             if repo.groups_added: # already added the groups from this repo
                 reposWithGroups.append(repo)
