diff -ru yum-3.2.29-orig/cli.py yum-3.2.29/cli.py
--- yum-3.2.29-orig/cli.py	2012-10-02 16:11:11.745351692 -0400
+++ yum-3.2.29/cli.py	2012-10-02 16:13:28.062367091 -0400
@@ -899,6 +899,7 @@
 
         oldcount = len(self.tsInfo)
 
+        done = False
         for arg in userlist:
             if (arg.endswith('.rpm') and (yum.misc.re_remote_url(arg) or
                                           os.path.exists(arg))):
@@ -908,6 +909,7 @@
 
             try:
                 self.reinstall(pattern=arg)
+                done = True
             except yum.Errors.ReinstallRemoveError:
                 self._checkMaybeYouMeant(arg, always_output=False)
             except yum.Errors.ReinstallInstallError, e:
@@ -925,6 +927,8 @@
                 self.verbose_logger.log(yum.logginglevels.INFO_2, e)
         if len(self.tsInfo) > oldcount:
             return 2, [_('Package(s) to reinstall')]
+        if not done:
+            return 1, [_('Nothing to do')]
         return 0, [_('Nothing to do')]
 
     def localInstall(self, filelist, updateonly=0):
